stan/math/prim/scal/prob/inv_gamma_rng.hpp

```diff
@@ -2,45 +2,58 @@
 #include <stan/math/prim/scal/err/check_positive_finite.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
-#include <stan/math/prim/scal/fun/lgamma.hpp>
-#include <stan/math/prim/scal/fun/gamma_q.hpp>
-#include <stan/math/prim/scal/fun/digamma.hpp>
-#include <stan/math/prim/scal/meta/length.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
 #include <stan/math/prim/scal/meta/VectorBuilder.hpp>
-#include <stan/math/prim/scal/meta/return_type.hpp>
-#include <stan/math/prim/scal/fun/grad_reg_inc_gamma.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
 #include <boost/random/gamma_distribution.hpp>
 #include <boost/random/variate_generator.hpp>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    inv_gamma_rng(double alpha,
-                  double beta,
-                  RNG& rng) {
+    /**
+     * Return a pseudorandom inverse gamma variate for the given shape
+     * and scale parameters using the specified random number generator.
+     *
+     * alpha and beta can each be a scalar, a std::vector, an Eigen::Vector, or
+     * an Eigen::RowVector. Any non-scalar inputs must be the same length.
+     *
+     * @tparam T_shape Type of shape parameter
+     * @tparam T_scale Type of scale parameter
+     * @tparam RNG type of random number generator
+     * @param alpha (Sequence of) positive shape parameter(s)
+     * @param beta (Sequence of) positive scale parameter(s)
+     * @param rng random number generator
+     * @return inverse gamma random variate
+     * @throw std::domain_error if alpha or beta are nonpositive
+     * @throw std::invalid_argument if non-scalar arguments are of different
+     * lengths
+     */
+    template <typename T_shape, typename T_scale, class RNG>
+    inline  typename VectorBuilder<true, double, T_shape, T_scale>::type
+    inv_gamma_rng(const T_shape& alpha, const T_scale& beta, RNG& rng) {
       using boost::variate_generator;
       using boost::random::gamma_distribution;
-
       static const char* function = "inv_gamma_rng";
 
       check_positive_finite(function, "Shape parameter", alpha);
       check_positive_finite(function, "Scale parameter", beta);
+      check_consistent_sizes(function, "Shape parameter", alpha,
+                             "Scale Parameter", beta);
 
-      variate_generator<RNG&, gamma_distribution<> >
-        gamma_rng(rng, gamma_distribution<>(alpha, 1 / beta));
-      return 1 / gamma_rng();
-    }
+      scalar_seq_view<T_shape> alpha_vec(alpha);
+      scalar_seq_view<T_scale> beta_vec(beta);
+      size_t N = max_size(alpha, beta);
+      VectorBuilder<true, double, T_shape, T_scale> output(N);
 
+      for (size_t n = 0; n < N; ++n) {
+        variate_generator<RNG&, gamma_distribution<> >
+          gamma_rng(rng, gamma_distribution<>(alpha_vec[n], 1.0 / beta_vec[n]));
```

Comment:
`1 / x` is preferred to `1.0 / x` if `x` isn't an integer.  Here both `1 / x` and `1.0 / x` are used in consecutive lines, which is confusing if they're intended to do the same thing.

Reply:
beta_vec[n] might be an int here is the issue. I switched to using static_cast<double> and 1 / x. Should be clearer what's going on now.